src/back/spv/writer.rs

```diff
@@ -1302,6 +1302,44 @@ impl Writer {
                         Some(crate::ScalarKind::Float) => spirv::Op::FMod,
                         _ => unreachable!(),
                     },
+                    crate::BinaryOperator::Equal => match left_ty_inner.scalar_kind() {
```

Comment:
since `None` is unreachable, we should just `unwrap()` here - this will also make the match exhaustive